ServiceWebsite/ServiceWebsite.UnitTests/Controllers/ParticipantControllerTests.cs

```diff
@@ -17,20 +18,73 @@ public class ParticipantControllerTests
         private const string Username = "some.user@hearings.reform.hmcts.net";
         private readonly Guid _hearingId = Guid.NewGuid();
         private ParticipantController _controller;
+        private Mock<IHearingsService> _hearingService;
+        private Mock<IParticipantService> _participantService;
 
         [SetUp]
         public void Setup()
         {
-            _controller = new ParticipantController();
+            _hearingService = new Mock<IHearingsService>();
+            _participantService = new Mock<IParticipantService>();
+            _controller = new ParticipantController(_hearingService.Object, _participantService.Object);
             _controller.MockUserIdentity(Username);
         }
 
+        [Test]
+        public async Task should_return_badrequest_if_hearingId_is_empty()
+        {
+            var result = await _controller.UpdateSuitabilityAnswers(Guid.Empty, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
+            var badRequestresult = (BadRequestObjectResult)result;
+            Assert.AreEqual(400, badRequestresult.StatusCode);
+        }
+
         [Test]
         public async Task should_return_not_found_if_no_hearing_for_user_is_found()
         {
-            var result = await _controller.UpdateSuitabilityAnswers(_hearingId, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
-            var noContentResult = (NoContentResult)result;
-            Assert.AreEqual(204, noContentResult.StatusCode);
+            // given service returns
+            _hearingService.Setup(x => x.GetParticipantIdAsync(Username, _hearingId))
+                .ThrowsAsync(new NotFoundException("message"));
+            var result = (NotFoundObjectResult)await _controller.UpdateSuitabilityAnswers(_hearingId, new System.Collections.Generic.List<HearingSuitabilityAnswer>());
+            Assert.AreEqual(404, result.StatusCode);
+        }
+
+        [Test]
+        public async Task should_return_unauthorized_for_hearing_the_user_is_not_participant_in()
+        {
+            // given service throws
+            _hearingService.Setup(x => x.GetParticipantIdAsync(Username, _hearingId)).Returns(Task.FromResult((Guid?)Guid.Empty));
```

Comment:
There's a method `ReturnsAsync` you can use as well, which looks a bit simpler

Reply:
Resolved